File: apps/scripts/restart_pod.py
# restart_pod.py
import argparse
from kubernetes import client, config

def restart_pod(namespace, pod_name):
    config.load_kube_config()
    v1 = client.CoreV1Api()

    print(f"Deleting pod '{pod_name}'in namespace '{namespace}'...")
    try:
        v1.delete_namespaced_pod(name=pod_name, namespace=namespace)
        print("✅ Pod restart triggered successfully.")
    except Exception as e:
        print(f"❌ Failed to restart pod: {e}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--namespace", required=True, help="Kubernetes namespace")
    parser.add_argument("--pod", required=True, help="Pod name")
    args = parser.parse_args()

    restart_pod(args.namespace,args.p)

def restart_pod(namespace, pod_name):
    config.load_kube_config()
    v1 = client.CoreV1Api()

    print(f"Deleting pod '{pod_name}'in namespace '{namespace}'...")
    try:
        v1.delete_namespaced_pod(name=pod_name, namespace=namespace)
        print("✅ Pod restart triggered successfully.")
    except Exception as e:
        print(f"❌ Failed to restart pod: {e}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--namespace", required=True, help="Kubernetes namespace")
    parser.add_argument("--pod", required=True, help="Pod name")
    args = parser.parse_args()

    restart_pod(args.namespace,args.pod)

File: apps/scripts/restart_pod.go
// restart_pod.go
package main

import (
	"flag"
	"fmt"
	"log"
	"path/filepath"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/kubernetes"
	"k8s.io/client-go/tools/clientcmd"
)

func main() {
	namespace := flag.String("namespace", "default", "Kubernetes namespace")
	podName := flag.String("pod", "", "Pod name")
	flag.Parse()

	if *podName == "" {
		log.Fatal("❌ Pod name is required using --pod")
	}

	kubeconfig := filepath.Join(homeDir(), ".kube", "config")
	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
	if err != nil {
		log.Fatalf("❌ Error loading kubeconfig: %v", err)
	}

	clientset, err := kubernetes.NewForConfig(config)
	if err != nil {
		log.Fatalf("❌ Error creating clientset: %v", err)
	}

	fmt.Printf("🔁 Restarting pod: %s in namespace: %s\n", *podName, *namespace)
	err = clientset.CoreV1().Pods(*namespace).Delete(*podName, metav1.DeleteOptions{})
	if err != nil {
		log.Fatalf("❌ Error deleting pod: %v", err)
	}
	fmt.Println("✅ Pod restart triggered successfully.")
}

func homeDir() string {
	return filepath.Dir("/root")
}
